File: core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/promisehandlers/InitializePortMappingHandler.scala
package edu.uci.ics.amber.engine.architecture.pythonworker.promisehandlers

import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand
import edu.uci.ics.amber.engine.common.virtualidentity.LinkIdentity
import edu.uci.ics.texera.workflow.common.tuple.schema.Schema

import scala.collection.mutable

object InitializePortMappingHandler {
  final case class InitializePortMapping(inputToOrdinalMapping: Map[LinkIdentity, (Int, String)], outputToOrdinalMapping: Map[LinkIdentity, (Int, String)])
    extends ControlCommand[Unit]
}

File: core/amber/src/main/python/core/architecture/handlers/initialize_port_mapping_handler.py
from proto.edu.uci.ics.amber.engine.architecture.worker import InitializePortMappingV2
from .handler_base import Handler
from ..managers.context import Context


class InitializePortMappingHandler(Handler):
    cmd = InitializePortMappingV2

    def __call__(self, context: Context, command: cmd, *args, **kwargs):
        context.port_manager.set_input_ports(command.input_to_ordinal_mapping)
        context.port_manager.set_input_ports(command.output_to_ordinal_mapping)
        return None
